### PR TITLE
ML integration

### PR DESCRIPTION
**Pull request by slimwizard**
Integrating the ML model into the API pipeline. The endpoint now receives an image file via POST request, extracts each spot and makes predictions, and ultimately updates the DB with spot availability. I also had to cut out the bottom two spots on the Nethken A map because they no longer accurately correlated with the actual parking spots from the camera view. Updated Receive Image test for new image, also added a test for a wrong key being passed with the file.

### CODE REVIEW

File: lot-availability-api/application.py [hunk 1]
```diff
@@ -71,29 +72,13 @@ def get_lots_by_location(lat_long):
...
-    sharp.save('../image-processing-server/tmp', format='PNG')
-
 def update_db_upon_rec(spot_num, lot_id, occ):
     row_changed = db.session.query(Spots).filter_by(spot_number=spot_num,
             lot_id=lot_id).update(dict(occupied=occ))
     db.session.commit()
     print('Spot {} occupied updated to {}.'.format(spot_num, occ))
 
-@application.route('/smart-lot/upload/<string:lot_id>/<string:key>', methods=['POST'])
+@application.route('/api/upload/<string:lot_id>/<string:key>', methods=['POST'])
```
Comment (grant-freeman) [inline lot-availability-api/application.py:81]: does this change not break the test?

File: testing/python/receive_image_test.py [hunk 2]
```diff
@@ -16,23 +16,34 @@
 
 class TestReceiveImage(unittest.TestCase): 
     occ = [i.occupied for i in application.get_all_rows(Spots)[::-1]]
-    output = subprocess.Popen(['curl', '-i', '-X', 'POST',
-        '-F', 'file=@{}'.format(os.path.abspath('../resources/test_image.png')),
+    output1 = subprocess.Popen(['curl', '-i', '-X', 'POST',
+        '-F', 'file=@{}'.format(os.path.abspath('../resources/api_test2.png')),
         'localhost:5000/smart-lot/upload/a19f71fc-4d20-4790-9e38-31df6a02ac76/shoop'],
```
Comment (grant-freeman) [inline testing/python/receive_image_test.py]: this url still says smart-lot while the route in application.py referes to /api/

File: lot-availability-api/application.py (hunk 1 above)
Comment (slimwizard) [reply lot-availability-api/application.py:81]: good catch. fixed.

File: testing/python/receive_image_test.py (hunk 2 above)
Comment (slimwizard) [reply testing/python/receive_image_test.py]: pushed fix, good catch.

**Review by grant-freeman — APPROVED** [review]
alright this looks good to me now

File: testing/python/receive_image_test.py [hunk 3]
```diff
@@ -16,23 +16,34 @@
 
 class TestReceiveImage(unittest.TestCase): 
     occ = [i.occupied for i in application.get_all_rows(Spots)[::-1]]
```
Comment (grant-freeman) [inline testing/python/receive_image_test.py:18]: actually, there's no longer a get_all_rows function either

**Pull request merged**